File: python/p12.py
# Problem 12
import math
triangle=0
i=1
while True:
    #print(triangle)
    triangle+=i
    i+=1
    count=0
   # j=1
    for j in range(1,int(math.sqrt(triangle))):
        #if i>triangle:
        #    break
        if triangle%j==0:
            count+=2
        j+=1
    if count>499:
        print("winner:",triangle)
        break

File: ruby/p12.rb
require 'mathn'

def count_divisors(x)
  count = 1
  (2..Math.sqrt(x)).each { |n| count += 1 if x % n == 0 }
  count *= 2 if x > 3
  count
end


current_triangle = 0
i = 1
loop do
  current_triangle += i
  total = count_divisors(current_triangle)
  if  total > 500
    puts current_triangle
    break
  end
  i += 1
end

